src/components/core/Button.tsx

```diff
@@ -222,25 +133,24 @@ const Button: FC<ButtonProps> = ({
         ...rest,
       };
 
-  // can always use the bootstrap button internally
+  const Link = React.forwardRef((props, ref) => <Component ref={ref} to={to} {...props} />);
+
   return (
-    <Component
-      className={clsx(
-        styles.button,
-        styles[variant],
-        inset && 'inset',
-        small && 'small',
-        block && 'block',
-        disabled && styles.disabled,
-        className
-      )}
+    <MuiButton
+      className={clsx(className, inset && 'inset', small && 'small', block && 'block')}
+      classes={{
+        outlined: styles[variant],
+      }}
+      component={Link}
+      variant="outlined"
+      color={variant === 'primary' ? 'primary' : 'default'}
       {...props}
     >
       <Typography variant="button" color="inherit" weight="boldLight">
```

Comment:
We need to drop the typography component inside the material button as well as the children. The material-ui buttons have their own props for passing icons - 
```jsx
<Button
  variant="contained"
  color="secondary"
  className={classes.button}
  startIcon={<DeleteIcon />}
>
  Delete
</Button>
```

Reply:
I also think this way but changing that requires changes in 87 places of reference.
I think a compromise would be to support both of the approches and do the changes in a seperate story because it requires time and testing.